.github/workflows/ci.yml

```diff
@@ -50,9 +50,29 @@ jobs:
         - windows: py39
         - linux: py38-oldestdeps
 
+  docs:
+    needs: [core]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.8'
+      submodules: false
+      pytest: false
+      toxdeps: tox-pypi-filter
+      cache-path: |
+        docs/_build/
+        docs/generated/
+        .tox/sample_data/
```

Comment:
Can we only restore one cache per build? Having this cached like this means that the second build doesn't need to download it, but ideally  the first build could restore it from a cache as well.

Reply:
Our reusable workflow only supports one cache per build. Otherwise we would need GitHub Actions to support `for` loops, or we could implement a fixed number of caches with `cache1-path`, `cache2-path`, and `cache3-path` etc. instead of just the one `cache-path`.

However, we could do:
```yaml
cache-path: |
  docs/_build/
  docs/generated/
  .tox/sample_data/
cache-key: docs-${{ github.run_id }}
cache-restore-keys: docs-
```

AFAIK in the first docs build this will restore the most recent cache with a key matching `docs-*` and then save a new cache with the `docs-${{ github.run_id }}` at the end of that build. Then in the docs gallery build it would restore `docs-${{ github.run_id }}`. I really don't think we should do this as we want to build the docs with clean `_build/` and `generated/` directories, and we can't restore the `sample_data/` directory only from the less specific cache.

Reply:
In the Azure template we could just iterate through a YAML list 😞 https://openastronomy-azure-pipelines.readthedocs.io/en/latest/run_tox_env.html#caching

Reply:
Yeah well BOOOoooOoOOoo